services/api/service.go

```diff
@@ -493,6 +527,85 @@ func (api *RelayAPI) startValidatorRegistrationDBProcessor() {
 	}
 }
 
+// simulateBlock sends a request for a block simulation to blockSimRateLimiter.
+func (api *RelayAPI) simulateBlock(ctx context.Context, opts blockSimOptions) (requestErr, validationErr error) {
+	t := time.Now()
+	requestErr, validationErr = api.blockSimRateLimiter.Send(ctx, opts.req, opts.isHighPrio, opts.fastTrack)
+	log := opts.log.WithFields(logrus.Fields{
+		"duration":   time.Since(t).Seconds(),
+		"numWaiting": api.blockSimRateLimiter.CurrentCounter(),
+	})
+	if validationErr != nil {
+		ignoreError := validationErr.Error() == ErrBlockAlreadyKnown || validationErr.Error() == ErrBlockRequiresReorg || strings.Contains(validationErr.Error(), ErrMissingTrieNode)
+		if !ignoreError {
+			// Mark builder as non-optimistic.
+			opts.builder.status.IsOptimistic = false
+			log.WithError(validationErr).Warn("block validation failed")
+			return nil, validationErr
+		}
+		log.WithError(validationErr).Warn("block validation failed with an ignorable error")
```

Comment:
now this will log both "validation failed" and "block validation successful".

was block validation successful? should it actually log both?

Reply:
changes the message to `ignorable error during block simulation` and then just return there. that way we just get one log per block simulation!